### PR TITLE
Add ProtectedHandler

### PR DESCRIPTION
**Pull request by arkn98**
This PR just has a basic implementation of a role-based `ProtectedHandler`.

### CODE REVIEW

File: internal/controller/holiday_controller.go [hunk 1]
```diff
@@ -50,34 +50,37 @@ func CreateHoliday() http.HandlerFunc {
...
-		response, err := holidaySvc.CreateHoliday(holiday)
-		if err != nil {
-			util.RespondWithError(w, err.(util.ResponseError))
-			return
-		}
-
-		w.Header().Set("Location", "/holidays/"+string(response.ID))
-		util.RespondWithJSON(w, http.StatusCreated, response)
+	return &ProtectedHandler{
+		PermissionName: "CREATE_HOLIDAYS",
```
Comment (dyaneshvarun) [inline internal/controller/holiday_controller.go:54]: How is this tied to a database !? We should look out in that aspect as well, how are we gonna set this up in database ?! 

Two Approaches here for creating the string : 

- Tie it up with CRUD + Model Name handlers
- Or Create unique identifiers for each feature rolled out.

Anyways we should try to pump in these metrics dynamically into the database if not present in the database and not just have a check at the code level. But from the API front, this looks cool.

Comment (arkn98) [reply internal/controller/holiday_controller.go:54]: We'll have a middleware before all of these routes, that will get the roles (eventually the permissions) the user has from the database and drop it inside `r.Context()`. We can then get the roles from the context and check if the permission mentioned for the controller is present in them.

Comment (dyaneshvarun) [reply internal/controller/holiday_controller.go:54]: For that purpose , we should have some permission manager right that will talk to the DB and stuff. And is this middleware to hit every time ?!

Comment (arkn98) [reply internal/controller/holiday_controller.go:54]: I currently don't have anything in mind right now. Let me come back after working it out.

**Pull request merged**